### DISCUSSION TITLE
Can burn execute ptl files for inference?

### DISCUSSION THREAD

**Discussion by knma1992** [tracel-ai/burn#1743] [Q&A]
I am using the flutter rust bridge, and I am experimenting with implementing my tensorflow and pytorch models in my rust code, so I can call my models from the same codebase without using any flutter packages that are platform specific. My tensorflow model is onnx convertible so I might get lucky to use it in burn. However, for my pytorch mobile (.ptl) models, I would love to just be able to call them using burn for inference.

**Comment by laggui** [comment]
You can import your pytorch weights with the [`PyTorchFileRecorder`](https://burn.dev/book/import/pytorch-model.html), but I'm not sure that works for pytorch mobile models.

To be honest, I am not too familiar with the ptl model format. After a quick glance at the docs it seems that it is a scripted module that is saved for the lite interpreter. Our `PyTorchFileRecorder` just needs the state_dict, but it requires an equivalent model to be defined in Burn so that the weights can be loaded in the correct modules.
